src/main/java/Parser.java

```diff
@@ -0,0 +1,218 @@
+                        }
+                    } catch (NumberFormatException e) {
+                        System.out.print("Invalid task number entered\n");
+                    } catch (IndexOutOfBoundsException e) {
+                        System.out.print("Task does not exist! Type 'list' to check task numbers\n");
+                    }
+                    Storage.saveTaskList(existingTaskList);
+                    return 1;
+
+                case "delete":
+                    try {
+                        if (splitInput[1].equalsIgnoreCase("all")) {
+                            existingTaskList.deleteAllTasks();
+                        }
+                        else {
+                            int taskNum = Integer.parseInt(splitInput[1]);
+                            existingTaskList.deleteTask(taskNum);
+                        }
+                    } catch (NumberFormatException e) {
+                        System.out.print("Invalid task number entered\n");
+                        System.out.print(dividerText);
+                    } catch (IndexOutOfBoundsException e) {
+                        System.out.print("Task does not exist! Type 'list' to check task numbers\n");
+                        System.out.print(dividerText);
+                    }
+                    Storage.saveTaskList(existingTaskList);
+                    return 1;
+
+                case "todo":
+                    String todoDescription = input.replace("todo", "").trim();
+                    if (todoDescription.equals("")) { // missing description
+                        System.out.print("Missing task description!\n");
+                        System.out.print(dividerText);
+                    } else if (todoDescription.contains("/by")) { // unnecessary due date
+                        System.out.print("Todo task cannot have a due date!\n");
+                        System.out.print(dividerText);
+                    } else if (todoDescription.contains("/from") ||
+                            todoDescription.contains("/to")) { // unnecessary from and to date
+                        System.out.print("Todo task cannot have a from and to date!\n");
+                        System.out.print(dividerText);
+                    } else {
+                        existingTaskList.addTodo(todoDescription);
+                    }
+                    Storage.saveTaskList(existingTaskList);
+                    return 1;
+
+                case "deadline":
+                    String deadlineDescriptionDueDate = input.replace("deadline", "").trim();
+                    if (!deadlineDescriptionDueDate.contains(" /by ")) { // missing '/by' keyword
+                        System.out.print("Missing due date!\n");
+                        System.out.print(dividerText);
+                        Storage.saveTaskList(existingTaskList);
+                        return 1;
+                    }
```

Comment:
Can consider leaving a blank line after the line to enhance the readability.

"Logical units within a block should be separated by one blank line." - Java coding standard website.